rdkit/Chem/RegistrationHash.py

```diff
@@ -0,0 +1,502 @@
+    This assumes that the ordering of those lists is not important
+    """
+    if sg.GetProp("TYPE") != "COP":
+        return None
+    ats = tuple(atRanks[x] for x in sg.GetAtoms())
+    if sortAtomAndBondOrder:
+        ats = tuple(sorted(ats))
+    props = sg.GetPropsAsDict()
+    # we are intentionally not doing the label here. Marvin adds one, biovia draw does not
+    res = dict(type="COP", atoms=ats, index=props.get("index", 0))
+    return res
+
+
+def _CanonicalizeSGroups(mol, dataFieldNames=None, sortAtomAndBondOrder=True):
+    """
+    NOTES: if sortAtomAndBondOrder is true then the atom and bond lists will be sorted.
+    This assumes that the ordering of those lists is not important
+    """
+    dataFieldNames = dataFieldNames or ["Atrop"]
+    atRanks, bndOrder = _GetCanonicalAtomRanksAndBonds(mol)
+    res = []
+    for sg in Chem.GetMolSubstanceGroups(mol):
+        lres = None
+        if sg.GetProp("TYPE") == "DAT":
+            lres = _CanonicalizeDataSGroup(sg, atRanks, bndOrder,
+                                            dataFieldNames,
+                                            sortAtomAndBondOrder)
+        elif sg.GetProp("TYPE") == "SRU":
+            lres = _CanonicalizeSRUSGroup(mol, sg, atRanks, bndOrder,
+                                           sortAtomAndBondOrder)
+        elif sg.GetProp("TYPE") == "COP":
+            lres = _CaonicalizeCOPSGroup(sg, atRanks, sortAtomAndBondOrder)
+
+        if lres is not None:
+            res.append(lres)
+    if len(res) > 1:
+        # we need to sort them, but dicts don't let you do that directly:
+        tres = sorted(tuple(x.items()) for x in res)
+        res = tuple(dict(x) for x in tres)
+        # update "index" and "PARENT" props
+        idxmap = {}
+        for i, itm in enumerate(res):
+            if "index" in itm:
+                idxmap[itm["index"]] = i + 1
+                itm["index"] = i + 1
+        for i, itm in enumerate(res):
+            if "PARENT" in itm:
+                itm["PARENT"] = idxmap[itm["PARENT"]]
+
+    return json.dumps(res)
+
+
+class EnhancedStereoUpdateMode(enum.Enum):
+    ADD_WEIGHTS = enum.auto()
+    REMOVE_WEIGHTS = enum.auto()
+
+
+def _UpdateEnhancedStereoGroupWeights(mol, mode):
+    if mode == EnhancedStereoUpdateMode.ADD_WEIGHTS:
+        factor = 1
+    elif mode == EnhancedStereoUpdateMode.REMOVE_WEIGHTS:
+        factor = -1
+    else:
+        raise ValueError('Invalid Enhanced Stereo weight update mode')
+
+    isotopesModified = False
+    stgs = mol.GetStereoGroups()
+    for stg in stgs:
+        stgt = stg.GetGroupType()
+        weight = factor * ENHANCED_STEREO_GROUP_WEIGHTS[stgt]
+        for at in stg.GetAtoms():
+
+            # Make sure the isotope is reasonable and is not present in more than
+            # one stereo group, and we are not messing it up
+            isotope = at.GetIsotope()
+            if mode == EnhancedStereoUpdateMode.ADD_WEIGHTS and isotope >= 100:
```

Comment:
I believe that this check makes it impossible to generate hashes for molecules with isotope-labeled atoms from fairly large chunks of the periodic table (including Iodine). 
It would be nice to avoid adding this constraint; how about changing the constant here to 1000   (Or even higher) and adjusting the stereo weights accordingly (if required)?

In any case we should makes sure that the isotope regex is consistent with this limit (I think that the regex at the moment only allows up to 399, but I'm no good with regexes) and should document what the largest supported isotope is in case people are using odd isotopes for internal purposes.

Reply:
This only applies to the atoms that are actually chiral and in stereo groups - an unlikely situation for iodine!

That being said, I've bumped the threshold to 999.